contracts/utils/contracts/src/LibBytesRichErrors.sol

```diff
@@ -0,0 +1,59 @@
+/*
+
+  Copyright 2019 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.9;
+
+import "./LibRichErrors.sol";
+
+
+library LibBytesRichErrors {
+
+    using LibRichErrors for *;
+
+    enum InvalidByteOperationErrorCodes {
+        FromLessThanOrEqualsToRequired,
+        ToLessThanOrEqualsLengthRequired,
+        LengthGreaterThanZeroRequired,
+        LengthGreaterThanOrEqualsFourRequired,
+        LengthGreaterThanOrEqualsTwentyRequired,
+        LengthGreaterThanOrEqualsThirtyTwoRequired,
+        LengthGreaterThanOrEqualsNestedBytesLengthRequired,
+        DestinationLengthGreaterThanOrEqualSourceLengthRequired
+    }
+
+    // bytes4(keccak256("InvalidByteOperationError(uint8,uint256,uint256)"))
+    bytes4 internal constant INVALID_BYTE_OPERATION_ERROR_SELECTOR =
+        0x28006595;
+
+    // solhint-disable func-name-mixedcase
+    function InvalidByteOperationErrorRevert(
+        InvalidByteOperationErrorCodes errorCode,
+        uint256 endpoint,
```

Comment:
Is this the right name? I see a couple places where this is actually the start offset. Maybe this should just be a generic `offset`...

Reply:
I was using `endpoint` in the sense that they are `endpoints` of a line segment. So the start is the beginning endpoint and the end is the ending endpoint. This said, I feel like we've used `offset` in other cases and it's probably less confusing. I'll update